packages/next/src/shared/lib/get-img-props.ts

```diff
@@ -283,6 +283,11 @@ export function getImgProps(
     config = { ...c, allSizes, deviceSizes }
   }
 
+  if (config.loaderFile && typeof defaultLoader === 'undefined') {
```

Comment:
```suggestion
  if (!isDefaultLoader) {
```

Use the existing variable on line 299 below

Reply:
Actually, move this if statement into the `else` block on line 309

Reply:
`isDefaultLoader` flag is just used to check whether or not the user pass loader function to `loader` props or `config.images.loaderFile`.

```ts
// get-img-props.ts

let loader: ImageLoaderWithConfig = rest.loader || defaultLoader // undefined
...
const isDefaultLoader = '__next_img_default' in loader
```

But If the loader file export a named function and the user do not pass `loader` props, the value of `loader` above is `undefined` so the error occurs like this 👉
```
TypeError: Cannot use 'in' operator to search for '__next_img_default' in undefined
```

So `defaultLoader` should be some value: `defaultLoader` or `config.images.loaderFile`